fsspec/implementations/local.py

```diff
@@ -63,46 +63,41 @@ def glob(self, path, **kwargs):
         return super().glob(path, **kwargs)
 
     def info(self, path, **kwargs):
+        t = None
         if isinstance(path, os.DirEntry):
             # scandir DirEntry
             out = path.stat(follow_symlinks=False)
             link = path.is_symlink()
+            if link:
+                t = "link"
             if path.is_dir(follow_symlinks=False):
                 t = "directory"
             elif path.is_file(follow_symlinks=False):
                 t = "file"
-            else:
-                t = "other"
             path = self._strip_protocol(path.path)
         else:
             # str or path-like
             path = self._strip_protocol(path)
             out = os.stat(path, follow_symlinks=False)
             link = stat.S_ISLNK(out.st_mode)
             if link:
-                out = os.stat(path, follow_symlinks=True)
+                out = os.lstat(path)
+                t = "link"
             if stat.S_ISDIR(out.st_mode):
                 t = "directory"
             elif stat.S_ISREG(out.st_mode):
                 t = "file"
-            else:
-                t = "other"
         result = {
             "name": path,
             "size": out.st_size,
-            "type": t,
+            "type": t or "other",
             "created": out.st_ctime,
             "islink": link,
         }
         for field in ["mode", "uid", "gid", "mtime"]:
             result[field] = getattr(out, "st_" + field)
         if result["islink"]:
             result["destination"] = os.readlink(path)
-            try:
-                out2 = os.stat(path, follow_symlinks=True)
-                result["size"] = out2.st_size
-            except IOError:
-                result["size"] = 0
```

Comment:
Hm, could you elaborate on why we are removing this size? The size of the link itself (and not the file it points to) seems useless to me, unless I'm missing something.

Reply:
https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=info#fsspec.archive.AbstractArchiveFileSystem.info
Accordign to docs, `info` is supposed to return ls-like information - which returns link size. 
But I agree, that this is probably not that useful. I will roll it back

Reply:
> Returns a single dictionary, with exactly the same information as ls would with detail=True.

> The default implementation should calls ls and could be overridden by a shortcut. kwargs are passed on to `ls().

@pared There might be a misunderstanding here. The docs talk about `fs.ls`, not CLI ls. Unless I'm missing something.